### DISCUSSION TITLE
JS - Objects - Creating, Accessing and Iterating

### DISCUSSION THREAD

**Discussion by akash-coded** [akash-coded/mern#159] [Tasks]
# Task Description
You are tasked with creating a program that stores information about a movie in an object. Your program should allow the user to add, access, and delete information about the movie.

# Requirements

- The program should use an object to store information about a movie, including its title, director, release year, and rating (on a scale of 1-10).
- The program should use hard-coded values to store information about the movie in the object.
- The program should allow the user to access information about the movie using the console.log() function.
- The program should also allow the user to delete information about the movie using the delete keyword.
- The program should use a for...in loop to iterate through the keys and values of the object and output them using the console.log() function.

# Example Output
> Title: The Matrix
> Director: The Wachowskis
> Release Year: 1999
> Rating: 9
> After deleting rating: undefined
> Title: The Matrix
> Director: The Wachowskis
> Release Year: 1999

**Comment by basu-cse** [comment]
const filmObj = {
	Title: "Baahubali"
	Director: "S. S. Rajamouli"
	Release Year: 2015
	Rating: 9
}

for (const key in filmObj) {
  console.log(key, " : ", filmObj[key]); 
}

delete filmObj.rating;

for (const key in filmObj) {
  console.log(key, " : ", filmObj[key]); 
}

**Comment by Manpreet-web** [comment]
let movie = {
title: "Money Heist",
director: "Álex Pina",
release_year: 2017,
rating: 10,
}

for(let key in movie){
console.log(key, " : ", movie[key]);
The Output will //
Title: Money Heist
Director: Alex Pina
Release year: 2017
Rating: 10
}
delete movie.rating;

for(let key in movie){
console.log(key, " : ", movie[key]);
After deleting The Output will //
Title: Money Heist
Director: Alex Pina
Release year: 2017

}

**Comment by VishalShukla85** [comment]
let movie = {
     Title: "PM Modi",
     Director: "Omung Kumar",
     Release_Year: 2019,
     rating: 9
}
console.log(movie)
for(let key in movie){
    console.log(key, ":", movie[key])
}

delete movie.rating;

for(let key in movie){
    console.log(key, ":", movie[key])
}

**Comment by SathyaGanes** [comment]
Code

```
let movie = {
  title: "The Dark Knight",
  director: "Christopher Nolan",
  releaseYear: 2008,
  rating: 9
};


console.log("Title:", movie.title);
console.log("Director:", movie.director);
console.log("Release Year:", movie.releaseYear);
console.log("Rating:", movie.rating);


delete movie.rating;
console.log("After deleting rating:", movie);


console.log("Iterating through movie object:");
for (let key in movie) {
  console.log(`${key}: ${movie[key]}`);
}

```


Output:
Title: The Dark Knight
Director: Christopher Nolan
Release Year: 2008
Rating: 9
After deleting rating: {title: "The Dark Knight", director: "Christopher Nolan", releaseYear: 2008}
Iterating through movie object:
title: The Dark Knight
director: Christopher Nolan
releaseYear: 2008

**Comment by chukku10** [comment]
const filmObj = {
Title: "Varisu"
Director: "Vamshi Paidipally"
Release Year: 2023
Rating: 6
}

for (const key in filmObj) {
console.log(filmObj[key]);
}

delete filmObj.rating;

for (const key in filmObj) {
console.log(filmObj[key]);
}

**Reply by Sagar1006-dev** [reply]
let movie={
  Title: "mrs undercover",
  Director:"rajesh sharma",
  release_year:2023,
  rating:6,
};
for(let key in movie){
console.log(key,":" , movie[key]);}
delete movie.rating;
for(let key in movie)
console.log(key,":",movie[key]);

**Comment by sam2elect** [comment]
# CODE
```TS

type Movie = {
    title: string;
    director: string;
    releaseYear: number;
    rating: number;
  };
  
  const movie: Movie = {
    title: "The Matrix",
    director: "The Wachowskis",
    releaseYear: 1999,
    rating: 9,
  };
  // Access information about the movie
  console.log("Title:", movie.title);
  console.log("Director:", movie.director);
  console.log("Release Year:", movie.releaseYear);
  console.log("Rating:", movie.rating);
  
  // Delete information about the movie
  delete (movie as any).rating;
  console.log("After deleting rating:", movie.rating);
  
  // Output all keys and values of the movie object
  for (const key in movie) {
    console.log(`${key}:`, movie[key]);
  }
  
```

**Comment by bipin-coded** [comment]
## Typescript Code

```typescript
let movie = {
  "titile": "After Dark in Central Park",
  "director": "XYZ",
  "year_of_release": 1900,
  "rating": 9
};

console.log(movie.titile);

for (let key in movie) {
  console.log(key, ' : ', movie[key]);
}

delete movie.rating;

for (let key in movie) {
  console.log(key, ' : ', movie[key]);
}
```

## Output
![Screenshot (3)](https://user-images.githubusercontent.com/126976134/232324165-52677530-a06d-4f76-a470-010a604af0fb.png)

**Comment by Ravi-710** [comment]
let movies={
   Title:"BatMan ",
   Directors:"abcs",
   Rating:"9",
   Release_year:"2018",

}
console.log(movies);
for(let key in movies){
    console.log(key, " : ", movies[key]);
    
    Title: BatMan
    Director: Abcs
    Release_year: 2018
    Rating: 9
    }
    delete movies.rating;
    
    for(let key in movies){
    console.log(key, " : ", movies[key]);

    Title:BatMan
    Director: Abcs
    Release_year: 2018
    
    }

**Comment by Prathmesh06** [comment]
```js
let movie = {
    Title : "3 idiots",
    Director : "rk hirani",
    ReleaseYear : 2009,
    Rating : 8.6,
    Afterdeletingrating: "undefined",
}

console.log(movie);

for (let key in movie) {
    console.log(key, " : ", movie[key]);
}

for (let key in movie) {
    delete movie[key]
}

console.log(movie);
```

![image](https://user-images.githubusercontent.com/58461727/232324236-46345360-cd11-4ab4-a5b7-54b84678ef9e.png)

**Comment by vk0027** [comment]
let movie = {
    title: "The Matrix",
    director: "The Wachowskis",
    releaseYear: 1999,
    rating: 9
  };
  
  console.log("Title:", movie.title);
  console.log("Director:", movie.director);
  console.log("Release Year:", movie.releaseYear);
  console.log("Rating:", movie.rating);
  
  delete movie.rating;
  
  for (let key in movie) {
    console.log(key,":", movie[key]);
  }

**Comment by saugatacr7** [comment]
let Movie = {
    name:'Mission Impossible',
    Director : 'Delhi',
    Release Year: '1996',
    Rating: '7.1',
}
for (let key in Movie) { 
    let value;
    value = Movie[key];

    console.log(key + " : " +  value)
}
delete Movie.Rating;
console.log(key + " : " +  value)

**Comment by RevathiSundaramoorthy** [comment]
## Code
```typescript
let movie_details: {Title: string, Director: string, "Release Year": number, Rating?: number} = {
    Title: "The Matrix",
    Director: "The Wachowskis",
    "Release Year": 1999,
    Rating: 9 
}
for (let key in movie_details) {
    console.log(key, " : ", movie_details[key]);
}
delete movie_details["Rating"];
console.log(movie_details["Rating"]);
for (let key in movie_details) {
    console.log(key, " : ", movie_details[key]);
}
```
## Screenshot
![image](https://user-images.githubusercontent.com/126975777/232328067-6b207173-ffbc-4148-a0d2-8eeb120f1628.png)

**Comment by starvikas1988** [comment]
var movie=[ {
    "Title": "The Matrix",
"Director": "The Wachowskis",
"Release Year": 1999,
Rating: 9
},
{
"Title": "The Matrix2",
"Director": "The Wachowskis2",
"Release Year": 1999,
}];

console.log(movie);

for(let key in movie){
    console.log(movie[key])
    for(let keyVal in movie[key]){
        console.log("key:"+keyVal+" , "+"value:"+movie[key][keyVal])
    }
}

**Comment by Ruchiarth** [comment]
```
```
let movie={
    title :"DDLJ ",
    director: "Yash Chopra",
    ReleaseYear:2000,
    Rating:10,


};
for (let key in movie){
    console.log(key,":",movie[key]);
}

delete movie.Rating;

for (let key in movie){
    console.log(key,":",movie[key]);
}

#screenshot
// Before : 
![image](https://user-images.githubusercontent.com/126975575/232324387-ab8bd1b7-78d1-4032-870d-249df02c256a.png)
// after
![image](https://user-images.githubusercontent.com/126975575/232324883-e0c318d2-2023-4f39-89e1-b0d43ddedc50.png)

**Comment by mownica-p** [comment]
# code
```typescript
// Create a movie object 
const movie = {
    title: "The Matrix",
    director: "The Wachowskis",
    releaseYear: 1999,
    rating: 9    
  };
  
  for (let key in movie){
    console.log(key,":",movie[key]);
    }
  
  // Delete rating information about the movie using delete keyword
  delete movie.rating;
  
  for (const key in movie) {
  console.log(key,":",movie[key]);
  }
  
```
# screenshot
![image](https://user-images.githubusercontent.com/126975660/233681101-2f8d3c05-8026-4844-8370-d787b106be77.png)

**Comment by sam2elect** [comment]
# CODE
```ts
interface Movie1 {
    title: string;
    director: string;
    releaseYear: number;
    rating: number;
  }
  
  const movie1: Movie1 = {
    title: "The Matrix",
    director: "The Wachowskis",
    releaseYear: 1999,
    rating: 9,
  };
  
  console.log(`Title: ${movie1.title}`);
  console.log(`Director: ${movie1.director}`);
  console.log(`Release Year: ${movie1.releaseYear}`);
  console.log(`Rating: ${movie1.rating}`);
  

  
  for (const key in movie1) {
    console.log(`${key}: ${movie1[key as keyof Movie1]}`);
  }

  delete (movie1 as any).rating;

  for (let key in movie1) {
    console.log(key, ' : ', movie1[key]);
  }
  
```
# SCREENSHORT

![image](https://user-images.githubusercontent.com/85131631/235346783-5e1858df-6780-4a06-bc40-b713e61d3a42.png)
![image](https://user-images.githubusercontent.com/85131631/235346795-7d26e3dd-68b4-42ed-b89e-b6242c49a794.png)

**Comment by kumarneeraj89** [comment]
# Code
```javascript
const Movie = {
   "Title": "The Matrix",
    "Director": "The Wachowskis",
    "Release Year": 1999,
    "Rating": 9
}

console.log(Movie);
delete Movie.Rating;
console.log(`After After deleting rating : ${Movie.Rating}`);
console.log(Movie);
```
# Screenshort

<img width="644" alt="image" src="https://user-images.githubusercontent.com/65106798/235444394-36c6855b-a9f0-458c-8768-0779c931643e.png">
